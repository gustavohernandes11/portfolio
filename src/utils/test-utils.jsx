import { render } from "@testing-library/react";
import { Theme } from "../theme/ThemeProvider";
import { GlobalContextProvider } from "../contexts/globalContext/GlobalContextProvider";

export const customRender = (ui) => {
    return render(
        <GlobalContextProvider>
            <Theme>{ui}</Theme>
        </GlobalContextProvider>
    );
};
export * from "@testing-library/react";

export { customRender as render };
