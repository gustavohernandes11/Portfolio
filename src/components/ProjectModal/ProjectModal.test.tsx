import { screen } from "@testing-library/react";
import { customRender } from "../../utils/test-utils";
import "@testing-library/jest-dom/extend-expect";
import "@testing-library/jest-dom";

import { ProjectModal } from "./index";

describe("<ProjectModal/>", () => {
    it("should render the ProjectModal component without buttons", () => {
        customRender(
            <ProjectModal
                projectName="projectNameText"
                description="descriptionText"
                imageAlt="altText"
                imageSrc="https://github.com/gustavohernandes11.png"
                placeholder="empty"
            />
        );
        const deployButton = screen.queryByRole("button", {
            name: /deploy/i,
        });
        const githubButton = screen.queryByRole("button", {
            name: /github/i,
        });

        expect(githubButton).not.toBeInTheDocument();
        expect(deployButton).not.toBeInTheDocument();
    });
    it("should render the ProjectModal with the buttons", () => {
        const { container } = customRender(
            <ProjectModal
                projectName="projectNameText"
                description="descriptionText"
                imageAlt="altText"
                imageSrc="https://github.com/gustavohernandes11.png"
                placeholder="empty"
                githubUrl="githubUrlText"
                deployUrl="deployUrlText"
            />
        );
        const deployButton = screen.queryByRole("button", {
            name: /deploy/i,
        });
        const githubButton = screen.queryByRole("button", {
            name: /github/i,
        });

        const projectModal = container.querySelector(
            "#sandbox > div > div > div"
        );

        expect(githubButton).toBeInTheDocument();
        expect(deployButton).toBeInTheDocument();
        expect(projectModal).toMatchSnapshot();
    });
});
