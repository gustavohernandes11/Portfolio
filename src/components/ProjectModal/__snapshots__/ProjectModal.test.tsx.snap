// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectModal/> should render the ProjectModal with the buttons 1`] = `null`;
